### PR TITLE
dialect: (riscv_snitch) prettify frep custom syntax

### PR DESCRIPTION
**Pull request by superlopuh**

### CODE REVIEW

File: xdsl/dialects/riscv_snitch.py [hunk 1]
```diff
@@ -161,28 +162,41 @@ def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
...
+
+        frep = cls(max_rep, body, IntAttr(stagger_mask), IntAttr(stagger_count))
+        if remaining_attributes is not None:
+            frep.attributes |= remaining_attributes.data
+        return frep
+
+    def print(self, printer: Printer) -> None:
+        printer.print_string(" ")
+        printer.print_ssa_value(self.max_rep)
+        if self.stagger_count.data and self.stagger_mask.data:
```
[the remark this answers is not in the document]
Comment (compor) [reply xdsl/dialects/riscv_snitch.py:188]: This got me thinking if having a zero mask and an actual stagger count does anything or if it's even legal.

**Review by compor — APPROVED** [review]

File: xdsl/dialects/riscv_snitch.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (superlopuh) [reply xdsl/dialects/riscv_snitch.py:188]: I bet it's legal and probably does nothing, a problem for the future :)

**Comment by tobiasgrosser** [comment]
This seems to be good-to-go. Should we merge it?

**Comment by superlopuh** [comment]
Thank you for the reminder

**Pull request merged**